r/R/compute.R

```diff
@@ -186,6 +186,32 @@ unique.ArrowDatum <- function(x, incomparables = FALSE, ...) {
   call_function("unique", x)
 }
 
+#' @export
+any.ArrowDatum <- function(..., na.rm = FALSE){
+  
+  a <- collect_arrays_from_dots(list(...))
+  result <- call_function("any", a)
+
+  if(!na.rm && a$null_count > 0 && !as.vector(result)){
+    Scalar$create(NA)
```

Comment:
```suggestion
  if (!na.rm && a$null_count > 0 && !as.vector(result)) {
    # Three-valued logic: with na.rm = FALSE, any(c(TRUE, NA)) returns TRUE but any(c(FALSE, NA)) returns NA
    # TODO: C++ library should take na.rm for any/all (like ARROW-9054)
    Scalar$create(NA)
```

FYI @rok, not sure if you want to add this to your PR for ARROW-9054 or if you'd like to do a followup to add any/all to it, up to you.

Reply:
> FYI @rok, not sure if you want to add this to your PR for ARROW-9054 or if you'd like to do a followup to add any/all to it, up to you.

I would maybe keep that as a separate followup PR, ARROW-9054 is already getting quite big ;)

Reply:
Agreed. Made [ARROW-12499](https://issues.apache.org/jira/browse/ARROW-12499) to capture this.